### PR TITLE
Add noMmap hint to ioHintSet

### PR DESCRIPTION
**Pull request by jeremiah-corrado**
Per the discussion in this issue: https://github.com/chapel-lang/chapel/issues/20346:

- `QIO_METHOD_PREADPWRITE` was added to the `ioHintSet` type as a new hint called `noMmap`
- The hidden `ioHintSet.direct` method was renamed to `ioHintSet.fromFlag()` as its name conflicted with the "direct" iohint that may be added at some point in the future.
- Switched uses of `ioHintSet.direct(QIO_HINT_PARALLEL)` to `ioHintSet.mmap`

Tests and library code were updated to reflect the above changes. 

- [x] paratest passing

### CODE REVIEW

File: modules/standard/IO.chpl [hunk 1]
```diff
@@ -1459,8 +1460,13 @@ record ioHintSet {
   */
   proc type mmap { return new ioHintSet(IOHINTS_MMAP); }
 
+  /* Suggests that 'mmap' should not be used to acces the file contents.
```
Comment (lydia-duncan) [inline modules/standard/IO.chpl:1463]: Typo: `acces`

File: test/release/examples/benchmarks/shootout/revcomp-fast.chpl [hunk 2]
```diff
@@ -11,7 +11,7 @@ const table = initTable("ATCGGCTAUAMKRYWWSSYRKMVBHDDHBVNN\n\n");
 proc main(args: [] string) {
   const stdin = openfd(0),
         input = stdin.reader(iokind.native, locking=false,
-                             hints=ioHintSet.direct(QIO_HINT_PARALLEL)),
+                             hints=ioHintSet.fromFlag(QIO_HINT_PARALLEL)),
```
Comment (lydia-duncan) [inline test/release/examples/benchmarks/shootout/revcomp-fast.chpl:14]: Should this be changed to use `ioHintSet.mmap` instead?  Or did you want to do that in a separate PR?

File: test/studies/shootout/reverse-complement/bharshbarg/revcomp-mark-skip-read-begin.chpl [hunk 3]
```diff
@@ -13,7 +13,7 @@ config const readSize = 16 * 1024;
 proc main(args: [] string) {
   const stdin = openfd(0);
   var input = stdin.reader(iokind.native, locking=false,
-                           hints=ioHintSet.direct(QIO_HINT_PARALLEL));
+                           hints=ioHintSet.fromFlag(QIO_HINT_PARALLEL));
```
Comment (lydia-duncan) [inline test/studies/shootout/reverse-complement/bharshbarg/revcomp-mark-skip-read-begin.chpl:16]: Ditto here

**Review by lydia-duncan — APPROVED** [review]
Looks good!  Just one typo and one question if some additional changes should be done now (and if so, you should include them in the PR message)

File: test/release/examples/benchmarks/shootout/revcomp-fast.chpl (hunk 2 above)
Comment (jeremiah-corrado) [reply test/release/examples/benchmarks/shootout/revcomp-fast.chpl:14]: Good point! I'll make that change in this PR.
